core/src/main/java/com/novoda/merlin/receiver/ConnectivityChangesRegister.java

```diff
@@ -8,42 +8,46 @@
 import android.os.Build;
 
 import com.novoda.merlin.service.AndroidVersion;
+import com.novoda.merlin.service.ConnectivityChangeEventExtractor;
 import com.novoda.merlin.service.MerlinService;
 
 public class ConnectivityChangesRegister {
 
     private final Context context;
     private final ConnectivityManager connectivityManager;
     private final AndroidVersion androidVersion;
+    private final ConnectivityChangeEventExtractor connectivityChangeEventExtractor;
 
     private ConnectivityReceiver connectivityReceiver;
     private ConnectivityCallbacks connectivityCallbacks;
 
     public ConnectivityChangesRegister(Context context,
                                        ConnectivityManager connectivityManager,
-                                       AndroidVersion androidVersion) {
+                                       AndroidVersion androidVersion,
+                                       ConnectivityChangeEventExtractor connectivityChangeEventExtractor) {
         this.context = context;
         this.connectivityManager = connectivityManager;
         this.androidVersion = androidVersion;
+        this.connectivityChangeEventExtractor = connectivityChangeEventExtractor;
     }
 
-    public void register(MerlinService.ConnectivityChangesListener connectivityChangesListener) {
+    public void register(MerlinService.ConnectivityChangesNotifier connectivityChangesNotifier) {
```

Comment:
Both `ConnectivityCallbacks` and `ConnectivityReceiverConnectivityChangeNotifier` now use the same interface.